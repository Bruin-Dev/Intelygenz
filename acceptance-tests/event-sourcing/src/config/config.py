# In order to work, this module must be executed in an environment with the environment variables referenced set.
# use source env in this directory.
# If you dont have any env files, ask for one they are not in VCS
import logging
import os
import sys

NATS_CONFIG = {
    "servers": [os.environ["NATS_SERVER1"]],
    "cluster_name": os.environ["NATS_CLUSTER_NAME"],
    "subscriber": {"pending_limits": 65536},
    "multiplier": 5,
    "min": 5,
    "stop_delay": 300,
    "reconnects": 150,
}

LOG_CONFIG = {
    "name": "base-nats-test",
    "level": logging.DEBUG,
    "stream_handler": logging.StreamHandler(sys.stdout),
    "format": "%(asctime)s: %(module)s: %(levelname)s: %(message)s",
}

REDIS = {"host": os.environ["REDIS_HOSTNAME"]}
