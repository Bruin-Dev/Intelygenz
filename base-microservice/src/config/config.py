# In order to work, this module must be executed in an environment with the environment variables referenced set.
# use source env in this directory.
# If you dont have any env files, ask for one they are not in VCS
import os

NATS_CONFIG = {
    'servers': [os.environ["NATS_SERVER1"]],
    'cluster_name': os.environ["NATS_CLUSTER_NAME"],
    'subscriber': {
        'max_inflight': 1,
        'pending_limits': 1
    },
    'publisher': {
        'max_pub_acks_inflight': 1
    }
}

LOG_CONFIG = {
    'name': 'velocloud-overseer-log',
}
