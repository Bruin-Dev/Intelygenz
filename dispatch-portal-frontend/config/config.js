export const config = {
  basePath: process.env.BASE_PATH,
  baseApi: process.env.BASE_API,
  environment: process.env.ENVIRONMENT,
  mocks: process.env.MOCKS,
  VENDORS: {
    LIT: 'LIT',
    CTS: 'CTS'
  }
};
