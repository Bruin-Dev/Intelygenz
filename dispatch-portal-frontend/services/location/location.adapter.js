export const locationInAdapter = data => ({
  owner: data.client_name,
  address1: data.client_address.address,
  country: data.client_address.country,
  city: data.client_address.city,
  state: data.client_address.state,
  zip: data.client_address.zip
});
