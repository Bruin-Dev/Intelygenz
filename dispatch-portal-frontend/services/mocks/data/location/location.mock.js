export const getLocationByTicketIdMock = {
  client_name: 'Red Rose',
  client_address: {
    address: '123 Fake Street',
    city: 'Pleasantown',
    state: 'New York',
    zip: '99085',
    country: 'United States'
  }
};
