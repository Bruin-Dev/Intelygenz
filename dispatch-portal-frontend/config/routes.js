import { config } from './config';

export const Routes = {
  BASE: () => `${config.basePath}`,
  DISPATCH: () => `${config.basePath}dispatch`,
  NEW_DISPATCH: () => `${config.basePath}new-dispatch`,
  LOGIN: () => `${config.basePath}login`
};
