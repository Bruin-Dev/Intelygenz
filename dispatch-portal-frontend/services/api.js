import axios from 'axios';
import { config } from '../config/config';

export const API_URLS = {
  LOGIN: '/login',
  DISPATCH: '/lit/dispatch',
  UPLOAD_FILES: '/upload'
};

export const baseConfig = {
  baseURL: config.baseApi,
  timeout: 10000,
  headers: {
    'Content-Type': 'application/json;charset=utf-8'
  }
};

export const axiosInstance = axios.create(baseConfig);
