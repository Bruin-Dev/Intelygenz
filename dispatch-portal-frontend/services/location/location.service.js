import { API_URLS } from '../api.config';
import axiosInstance from '../api';
import { locationInAdapter } from './location.adapter';

export class LocationService {
  constructor(axiosAuxI = axiosInstance) {
    this.axiosI = axiosAuxI;
  }

  async getLocationByTicketId(ticketId) {
    try {
      const res = await this.axiosI.get(
        `${API_URLS.GET_LOCATION_BY_TICKET_ID}/${ticketId}`
      );
      return locationInAdapter(res.data);
    } catch (error) {
      return this.captureErrorGeneric(error);
    }
  }

  // eslint-disable-next-line class-methods-use-this
  captureErrorGeneric(error) {
    return { error: error.message };
  }
}
