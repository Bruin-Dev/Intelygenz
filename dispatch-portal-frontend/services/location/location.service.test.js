import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { LocationService } from './location.service';
import { getLocationByTicketIdMock } from '../mocks/data/location/location.mock';
import { API_URLS } from '../api.config';

describe('location service tests', () => {
  const ticketIdMock = 'foo';
  const axiosIMocksTest = axios.create();
  let mockadapter;

  beforeEach(() => {
    mockadapter = new MockAdapter(axiosIMocksTest);
  });

  afterEach(() => {
    mockadapter.reset();
  });

  describe('getLocationByTicketId', () => {
    it('should match the snapshot', async () => {
      // Arrange.
      mockadapter
        .onGet(new RegExp(`${API_URLS.GET_LOCATION_BY_TICKET_ID}/*`))
        .reply(200, getLocationByTicketIdMock);

      // Act.
      const response = await new LocationService(
        axiosIMocksTest
      ).getLocationByTicketId(ticketIdMock);

      // Assert.
      expect(response).toMatchSnapshot();
    });

    it('should return an error if the service fail', async () => {
      // Arrange.
      mockadapter
        .onGet(new RegExp(`${API_URLS.GET_LOCATION_BY_TICKET_ID}/*`))
        .reply(404);

      // Act.
      const response = await new LocationService(
        axiosIMocksTest
      ).getLocationByTicketId(ticketIdMock);

      // Assert.
      expect(response).toMatchSnapshot();
    });
  });
});
