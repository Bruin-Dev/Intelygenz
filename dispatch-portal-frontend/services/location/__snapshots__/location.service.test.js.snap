// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`location service tests getLocationByTicketId should match the snapshot 1`] = `
Object {
  "address1": "123 Fake Street",
  "city": "Pleasantown",
  "country": "United States",
  "owner": "Red Rose",
  "state": "New York",
  "zip": "99085",
}
`;

exports[`location service tests getLocationByTicketId should return an error if the service fail 1`] = `
Object {
  "error": "Request failed with status code 404",
}
`;
